pybossa/cache/users.py

```diff
@@ -348,9 +348,21 @@ def delete_user_pref_metadata(name):
 
 @memoize(timeout=ONE_DAY)
 def get_user_preferences(user_id):
+    user_pref = get_user_pref(user_id)
+    user_email = get_user_email(user_id)
```

Comment:
single db query to obtain user object and extract user_pref and email_addr can be done here

Reply:
you can pass user.user_pref and user.email_addr ,
```
return get_user_pref_db_clause(user.user_pref, user.email_addr)
```

Reply:
I thought this will fail if the user object is empty.